resteasy-core/src/main/java/org/jboss/resteasy/core/ContextParameterInjector.java

```diff
@@ -125,6 +130,15 @@ public Object invoke(Object o, Method method, Object[] objects) throws Throwable
                }
                throw new LoggableFailure(Messages.MESSAGES.unableToFindContextualData(rawType.getName()));
             }
+            // Fix for RESTEASY-1721
+            if ("javax.servlet.http.HttpServletResponse".equals(rawType.getName()))
```

Comment:
This question is for my own personal knowledge. Is there a reason we use a proxy here?

As to this line, we need to think this through because this would not work in Jakarta EE 9.

Reply:
1. As to why these are proxies, that probably goes back to before my time, but I suppose it was just for convenience.

2. As to "javax", yeah, that's a good point. I don't want to assume the presence of servlet classes. Maybe
```
rawType.getName().contains("servlet.http.HttpServletResponse")
```
Well, actually, 
```
"javax.servlet.http.HttpServletResponse".equals(rawType.getName() || "jakarta.servlet.http.HttpServletResponse".equals(rawType.getName()
```
would be better, I think.

Reply:
Actually thinking about it, it probably doesn't matter. It's not like we can run with the namespace without changes anyway. I say leave it as is.

Reply:
That's true. And the new test will fail if there's no "javax.servlet.http.HttpServletResponse" on the classpath.